### DISCUSSION TITLE
Sort Order of Files with identical Date (from external Library)

### DISCUSSION THREAD

**Discussion by andbgr** [immich-app/immich#13120] [Q&A]
Hello,

i have in an external library, say, 5 Files (actually i have more examples of this):

1990 04 10 Birthday Party 01.jpg
1990 04 10 Birthday Party 02.jpg
1990 04 10 Birthday Party 03.jpg
1990 04 10 Birthday Party 04.jpg
1990 04 10 Birthday Party 05.jpg

they all have an identical Date (DateTimeOriginal is "1990:04:10 00:00:00" on each of them)

now, in the timeline, these images appear on the right date (that is not the issue) but, in (as far as i can tell) random order, and in different order on different clients (tested 2 phones and 1 web interface, 3 different results. they do appear in correct order in folder view in web interface)

is this expected behaviour?
are there secondary sort criteria (like filename...) if date is identical?
i found [this](https://github.com/immich-app/immich/pull/2119) pull request which seems to be related, but if i understand correctly it was never merged

note:
i am hesitant to post files, because of the personal content, but, if you say this is unexpected behaviour, i will find some.

regards,
Andi

**Comment by radh21301** [comment]
Hey. Not that your metadata is wrong, but I would suggest you fix the metadata instead if depending on immich to do it. It is also better if/when you move out of immich.

I used this cool command a while back when I was importing images to immich:
```exiftool '-AllDates+<0:0:${FileSequence;$_*=1}' -fileorder FileName ./```

It basicallt sorts the images in that folder by filename. and then adds a second increment to each of them..
File1 = 9AM 1sec
File2 = 9AM 2 sec etc.

will help solve your problem. And this will also be robust in the sense, you dont have to depend on the logic immich uses to sort images..DateTime Exif flag is always the priority
As always - please test these commands and do extra research before running on your library.

**Reply by andbgr** [reply]
Hello,
i had already considered something like this as a workaround, and i am still considering it if i find no other solution, but my issue is this: If i (or someone else) later look at "1990:04:10 00:00:00" we will (rightly) interpret this as a lack of accuracy ("there appears to be no time of day information") and i can later easily search for files with manually set date based on this, should i want to go back and add a more accurate date+time (my dad is the one responsible for the filenames, and sometimes he goes back and adds information later).
